test/integration/job/job_test.go

```diff
@@ -1521,87 +1497,66 @@ func TestParallelJobWithCompletions(t *testing.T) {
 	t.Cleanup(setDuringTest(&jobcontroller.MaxUncountedPods, 10))
 	t.Cleanup(setDuringTest(&jobcontroller.MaxPodCreateDeletePerSync, 10))
 	t.Cleanup(setDurationDuringTest(&jobcontroller.DefaultJobPodFailureBackOff, fastPodFailureBackoff))
-	cases := map[string]struct {
-		enableReadyPods bool
-	}{
-		"none": {},
-		"ready pods": {
-			enableReadyPods: true,
+	closeFn, restConfig, clientSet, ns := setup(t, "completions")
+	defer closeFn()
+	ctx, cancel := startJobControllerAndWaitForCaches(t, restConfig)
+	defer cancel()
+
+	jobObj, err := createJobWithDefaults(ctx, clientSet, ns.Name, &batchv1.Job{
+		Spec: batchv1.JobSpec{
+			Parallelism: pointer.Int32(54),
+			Completions: pointer.Int32(56),
 		},
+	})
+	if err != nil {
+		t.Fatalf("Failed to create Job: %v", err)
 	}
-	for name, tc := range cases {
-		t.Run(name, func(t *testing.T) {
-			defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.JobReadyPods, tc.enableReadyPods)()
-			closeFn, restConfig, clientSet, ns := setup(t, "completions")
-			defer closeFn()
-			ctx, cancel := startJobControllerAndWaitForCaches(t, restConfig)
-			defer cancel()
-
-			jobObj, err := createJobWithDefaults(ctx, clientSet, ns.Name, &batchv1.Job{
-				Spec: batchv1.JobSpec{
-					Parallelism: pointer.Int32(54),
-					Completions: pointer.Int32(56),
-				},
-			})
-			if err != nil {
-				t.Fatalf("Failed to create Job: %v", err)
-			}
-			want := podsByStatus{Active: 54}
-			if tc.enableReadyPods {
-				want.Ready = pointer.Int32(0)
-			}
-			validateJobPodsStatus(ctx, t, clientSet, jobObj, want)
+	want := podsByStatus{
+		Active: 54,
+		Ready:  pointer.Int32(0),
```

Comment:
```suggestion
		Ready:  pointer.Int32(52),
```

and remove the line below

Reply:
If we delete `validateJobPodsStatus` then `setJobPodsReady` will not have time to get the list of pods. I haven't dug deep, but it seems to me that we are not waiting to receive them and we need to add polling for that. If these changes make sense I can do it or we can leave it as is